rust/theoros/src/types/calldata.rs

```diff
@@ -35,72 +32,69 @@ pub struct Calldata {
 }
 
 impl Calldata {
-    pub async fn build_from(
-        state: &AppState,
-        chain_name: EvmChainName,
-        feed_id: String,
-    ) -> Result<Calldata, GetCalldataError> {
-        let event: DispatchUpdateInfos = state
-            .storage
-            .dispatch_events()
-            .get(&feed_id)
-            .await
-            .map_err(|_| GetCalldataError::DispatchNotFound)?
-            .ok_or(GetCalldataError::DispatchNotFound)?;
-
-        let validators = state
-            .hyperlane_validators_mapping
-            .get_validators(chain_name)
-            .ok_or(GetCalldataError::ChainNotSupported(format!("{:?}", chain_name)))?;
-
-        let checkpoints = state
-            .storage
-            .validators_checkpoints()
-            .get_validators_signed_checkpoints(validators, event.message_id)
-            .await
-            .map_err(|_| GetCalldataError::ValidatorNotFound)?;
-
-        // TODO: We only have one validator for now
-        let checkpoint_infos = checkpoints.last().unwrap();
-
-        let update = match event.update {
-            DispatchUpdate::SpotMedian { update, feed_id: _ } => update,
+    pub async fn build_from(state: &AppState, chain_name: EvmChainName, feed_id: String) -> anyhow::Result<Calldata> {
```

Comment:
we can use a builder pattern here

Reply:
We could but I'm not sure about the actual benefits since in practice we don't really "customize" how we call this function. But it would be prettier for sure

Reply:
easier to refactor in the future when there are more chain types and other stuff in the state

Reply:
also adds some default values which I think makes sense